### PR TITLE
correctly discover the Alertmanager cluster and ServiceMonitors

### PR DESCRIPTION
**Pull request by brancz**
Not sure how I missed these earlier but worth trying out I would say.

@fabxc @alexsomesan

### CODE REVIEW

File: manifests/examples/example-app/prometheus-frontend.yaml [hunk 1]
```diff
@@ -7,10 +7,9 @@ metadata:
     prometheus: frontend
 spec:
   version: v1.4.1
-  serviceMonitors:
-  - selector:
-      matchLabels:
-        tier: frontend
+  serviceMonitorSelector:
+    matchExpressions:
+    - {key: tier, operator: In, values: [frontend]}
```
Comment (fabxc) [inline manifests/examples/example-app/prometheus-frontend.yaml]: Any reason for changing to expression matching. In the general case plain label matching seems simpler.

**Comment by fabxc** [comment]
👍 

At some point we have to investigate what needs to be added to allow the k8s Prometheus to also be purely generated.

File: manifests/examples/example-app/prometheus-frontend.yaml (hunk 1 above)
Comment (brancz) [inline manifests/examples/example-app/prometheus-frontend.yaml]: fixed

**Comment by brancz** [comment]
Yes I was thinking of exactly the same thing. I will investigate tomorrow what we can get out of self-hosted today and what we still need.

**Comment by brancz** [comment]
@fabxc as a follow-up there will be a point where all components will be self-hosted, so we can already come up with changes that will replace this manually maintained prometheus config with a replacement using the `ServiceMonitor`. This will come relatively soon, what's left is finishing work on bootkube regarding self hosted etcd, flannel and improvements on checkpointing.

**Pull request merged**